vector/src/main/java/im/vector/app/features/location/live/map/LocationLiveMapViewModel.kt

```diff
@@ -70,7 +79,12 @@ class LocationLiveMapViewModel @AssistedInject constructor(
     }
 
     private fun handleStopSharing() {
-        locationSharingServiceConnection.stopLiveLocationSharing(initialState.roomId)
+        viewModelScope.launch {
+            val result = stopLiveLocationShareUseCase.execute(initialState.roomId)
```

Comment:
could the live location end up in a stuck state if the viewmodel was to close/cancel before the `stopLiveLocationShareUseCase` completed?

Reply:
I guess, it may happen the live is not stopped if `ViewModel` is cleared before the new state event is sent. When we are in this case, there will be no change on the UI and the service to send locations will still be running. So the user will hopefully see that it was not stopped. 

I see there is a `Session.coroutineScope` but not sure it is a good idea to use it here since we post a result ViewEvent in the coroutine.

Reply:
ahh does this mean the user will see a foreground notification showing that sharing is still a happening? 

if so it could be handy to support cancelling via the notification (as a separate feature)

Reply:
Yes, there will be the remaining foreground notification. I think we cannot really add a stop action on the notification since the user may be sharing its live location in several rooms. Or maybe we can decide via the Notification, it stops all current running lives. I will suggest the idea of such action.